evalml/pipelines/utils.py

```diff
@@ -1493,10 +1505,25 @@ def stack_X(X, series_id_name, time_index, starting_index=None):
                 starting_index=starting_index,
             ),
         )
-    restacked_X = pd.concat(restacked_X, axis=1)
-
     time_index_col = X[time_index].repeat(len(series_ids)).reset_index(drop=True)
-    time_index_col.index = restacked_X.index
-    restacked_X[time_index] = time_index_col
+
+    if len(restacked_X) == 0:
```

Comment:
can you help me understand why we need this logic block here? If `restacked_X` has length 0 why don't we error out?

Reply:
`restacked_X` can be zero in the case where we only have the time_index columns in the unstacked `X` dataframe. We want to be able to still stack in this case, which is why we have this logic block.

Reply:
That being said, with the refactor I just put in, I ended up condition to be if len(`original_columns`) == 0.